server/actions/InventoryItems.ts

```diff
@@ -0,0 +1,65 @@
+import InventoryItemSchema from 'server/models/InventoryItem'
+import * as MongoDriver from 'server/actions/MongoDriver'
+import { InventoryItem } from 'utils/types'
+import { ApiError } from 'utils/types'
+
+/**
+ * Checks to see if an item is in the inventory and will add to the quantity or
+ * create the item if needed.
+ * @param item the item that will be added (checkedIn) to db
+ * @param quantity of item to add
+ */
+export async function checkInInventoryItem(
+  item: InventoryItem,
+  quantityItem: number
+) {
+  const itemMatches = await MongoDriver.findEntities(InventoryItemSchema, item)
+  if (itemMatches.length) {
+    itemMatches[0].quantity += quantityItem
+    itemMatches[0].attributes?.sort()
+    MongoDriver.updateEntity(
+      InventoryItemSchema,
+      itemMatches[0].id,
+      itemMatches[0]
+    )
+  } else {
+    const createdItem = await MongoDriver.createEntity(
+      InventoryItemSchema,
+      item
+    )
+    createdItem.attributes?.sort()
+    MongoDriver.updateEntity(InventoryItemSchema, createdItem.id, createdItem)
```

Comment:
no need to update here since you will be sorting the attributes before creating.